rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -443,6 +443,16 @@ rmw_create_client(
   memcpy(const_cast<char *>(rmw_client->service_name), service_name, strlen(service_name) + 1);
 
   {
+    // It's unnecessary to lock `lck` in current scope.
+    // Unlock `lck` first and re-lock it after exiting the scope can avoid `lock-order-inversion`.
+    // https://github.com/ros2/ros2/issues/1488
+    lck.unlock();
+    auto lock_scope_exit = rcpputils::make_scope_exit(
+      [&lck]() {
+        // lock again for cleanup actions
+        lck.lock();
+      });
```

Comment:
this mechanism is required to go with current implementation. for example, if `rmw_fastrtps_shared_cpp::__rmw_publish` fails, entities need to be destroyed with this lock acquired.
Or probably we could have `rmw_fastrtps_shared_cpp::create/destroy_client` and acquire the lock `entity_creation_mutex_` inside of that only. the latter makes more code refactoring.